src/coredump/coredump.c

```diff
@@ -1656,28 +1660,35 @@ static int forward_coredump_to_container(Context *context) {
         return 0;
 }
 
-static int gather_pid_mount_tree_fd(const Context *context) {
+static int gather_pid_mount_tree_fd(const Context *context, int *ret_fd) {
         _cleanup_close_ int mntns_fd = -EBADF, root_fd = -EBADF;
         _cleanup_close_pair_ int pair[2] = EBADF_PAIR;
         int fd = -EBADF, r;
-        pid_t child;
 
         assert(context);
+        assert(ret_fd);
 
         /* Don't bother preparing environment if we can't pass it to libdwfl. */
 #if !HAVE_DWFL_SET_SYSROOT
-        return -EOPNOTSUPP;
+        r = 0;
+        goto finish;
 #endif
 
-        if (!arg_access_container)
-                return -EOPNOTSUPP;
+        if (!arg_access_container) {
+                r = 0;
+                goto finish;
+        }
 
-        if (socketpair(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0, pair) < 0)
-                return log_error_errno(errno, "Failed to create socket pair: %m");
+        if (socketpair(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0, pair) < 0) {
+                r = log_error_errno(errno, "Failed to create socket pair: %m");
+                goto finish;
```

Comment:
we have this rule in systemd: on success always initialize return params + on failure do not touch thim. In this error path you will touch *ret_fd  however

Reply:
or in other words, just drop this specific change?

Reply:
Unaddressed.